src/commands/utils/clusters.ts

```diff
@@ -575,3 +576,31 @@ function isDefinedManagedCluster(cluster: azcs.ManagedCluster): cluster is Defin
         cluster.nodeResourceGroup !== undefined
     );
 }
+
+export type Cluster = {
+    name: string;
+    clusterId: string;
+    resourceGroup: string;
+    subscriptionId: string;
+}
+
+export async function getClusters(sessionProvider: ReadyAzureSessionProvider, subscriptionId: string): Promise<Cluster[]> {
```

Comment:
suggestion here please to use javascript sdk:

```
export async function listManagedClusters(
    sessionProvider: ReadyAzureSessionProvider,
    subscriptionId: string,
): Promise<Errorable<Cluster[]>> {
    const client = getAksClient(sessionProvider, subscriptionId);
    let clusters: Cluster[] = [];
    try {
        const managedClustersList = client.managedClusters.list();
        for await (const cluster of managedClustersList) {
            clusters.push({
                name: cluster.name!,
                clusterId: cluster.id!,
                resourceGroup: cluster.nodeResourceGroup!, // if you need resourceGroup you can pass it in or parse it from resourceID
                subscriptionId: subscriptionId,
            });
        }

        return { succeeded: true, result: clusters };
    } catch (e) {
        return { succeeded: false, error: `Failed to retrieve list of managed cluster for ${subscriptionId}: ${e}` };
    }
}
```

Reply:
Thanks Hari, Can make the improvements in follow up PR